text/0000-buildpacks-config.md

```diff
@@ -0,0 +1,314 @@
+    - cnbs/sample-app
+    - 'cnbs/sample-app:v1'
+    - 'grc.io/cnbs/sample-app:v1'
+  builder:
+    name: 'cnbs/sample:builder'
+    trusted: true
+  previous:
+    name: cnbs/sample-app
+  run:
+    name: 'cnbs/sample-stack-run:bionic'
+    mirrors:
+      - 'grc.io/cnbs/sample-stack-run:bionic'
+env:
+  - name: ENV_1
+    value: ENV_2
+    operation: append
+buildpacks:
+  - id: samples/hello-world
+    version: 0.0.1
+  - id: samples/java-maven
+    version: 0.0.1
+source:
+  workspace: /workspace
+  include:
+    - cmd/
+    - go.mod
+    - go.sum
+    - '*.go'
+cache:
+  format: image
+  name: 'cnbs/sample-app-cache:build'
+process:
+  default: web
+  web:
+    env:
+      - name: PATH
+        value: ENV_2
+        operation: append
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+### Configuration Properties
+
+To prevent unnecessary complexity for end-users (app developers), the `buildpacks.<ext>` file would only support properties associated with Cloud Native Buildpacks. See full list of properties in the [file](#file) section.
+
+### Cross-Platform Support
+
+Support for this configuration would be part of a _new_ specification (`Application API`). 
+
+Building upon [RFC PR#182][rfc-pr-182], the `buildpacks.<ext>` file would be part of a contract between the `application` -> `converter` -> `platform`. Platforms will be required to support the `buildpacks.<ext>` files through changes to the `Platform API` and use of an intermediate file named _platform config_ (`platform.toml`). The `platform.toml` will be a Platform API version specific configuration file which contains only properties the platform can handle. Any unsupported features should yield a warning. Platforms can additional opt-out of features but would also be requested to yield warnings or errors in such cases.
+
+```text
+                      Application API
+┌────────────────────────────────────────────────────────────────┐
+│                                                                │
+
+                         ┌───────┐
+                         │       │   buildpacks.*    ┌───────────┐
+┌────────┐               │   p   ├───────────────────►           │
+│        │ buildpacks.*  │   l   │                   │ converter │
+│ source ├───────────────►   a   │                   │           │
+│        │               │   t   │                   └──┬────────┘
+└────────┘               │   f   │                      │
+                         │   o   ◄──────────────────────┘
+                         │   r   │   platform.toml
+                         │   m   │
+                         │       │
+                         └───────┘
+
+                         │                                       │
+                         └───────────────────────────────────────┘
+                                       Platform API
+```
+
+
+[rfc-pr-182]: https://github.com/buildpacks/rfcs/pull/182
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
```

Comment:
- `buildpacks.*` is easily confused with `buildpack.toml`
- `buildpacks*` is not inherently portable. We still need to implement a mechanism and more spec to convert it to something portable
- yet-another-config-file: I won't have a place to put config that's common across tools (license, include/exclude, version, etc)
- doesn't map well to pack commands/flags